aptos-move/move-examples/marketplace/sources/coin_listing.move

```diff
@@ -374,24 +340,24 @@ module coin_listing {
         object: Object<Listing>,
     ) acquires FixedPriceListing {
         let expected_seller_addr = signer::address_of(seller);
-        let (actual_seller_addr, _fee_schedule) = listing::close(object, expected_seller_addr);
+        let (actual_seller_addr, fee_schedule) = listing::close(object, expected_seller_addr);
         assert!(expected_seller_addr == actual_seller_addr, error::permission_denied(ENOT_SELLER));
 
         let listing_addr = object::object_address(&object);
         assert!(exists<FixedPriceListing<CoinType>>(listing_addr), error::not_found(ENO_LISTING));
         let FixedPriceListing {
             price: _,
-            purchase_event,
         } = move_from<FixedPriceListing<CoinType>>(listing_addr);
 
-        let purchase_event_data = PurchaseEvent {
-            purchaser: expected_seller_addr,
-            price: 0,
-            commission: 0,
-            royalties: 0,
-        };
-        event::emit_event(&mut purchase_event, purchase_event_data);
-        event::destroy_handle(purchase_event);
+        // TODO: Do we emit a different event for ending fixed price
+        fee_schedule::emit_purchase_event(
+            fee_schedule,
+            listing_addr,
```

Comment:
When the listing gets deleted, you don't know which token the listing is for. Can we include token address here too?